Added Python Solution for MooBuzz (Silver) (#1943)
cpinitiative/usaco-guide: solutions/usaco-966.mdx

---
id: usaco-966
source: USACO Silver 2019 December
title: MooBuzz
author: Ryan Chou
---

> [Official Analysis (C++)](http://www.usaco.org/r/data/sol_moobuzz_silver_dec19.html)

Reviewer: ```suggestion
[Official Analysis (C++)](http://www.usaco.org/current/data/sol_moobuzz_silver_dec19.html)
```

> For every 15 numbers, there are only 8 numbers spoken. (see image below)

Reviewer 2: ```suggestion
For every 15 numbers, there are only 8 numbers where the cows moo. (see image below)
```

> ![](/solutions/USACO-dec19-moobuzz-silver/first30_relation.png)

Reviewer: ```suggestion
![](USACO-dec19-moobuzz-silver/first30_relation.png)
```
i think the path should be like this?

Author: oops forgot to remove that :/

> Therefore, we can just get the distance away $N$ is from the first $8$ by calculating $\lfloor N / 8 \rfloor \times 15$, since we have to add $15$ for every grouping of $15$. 

Reviewer: ```suggestion
Therefore, we can just get the distance away $N$ is from the first $8$ by calculating  $\lfloor \frac{N}{8} \rfloor \times 15$, since we have to add $15$ for every grouping of $15$. 
```

Now it's just a matter of adding the corresponding first $8$ to the answer. 

## Implementation

**Time Complexity:** $ \mathcal{O}(1)$
<LanguageSection>
<PySection>
```py
import sys

sys.stdin = open('moobuzz.in', 'r')
sys.stdout = open('moobuzz.out', 'w')

N = int(input())

# Lists the first 8. 
# 14 is in the zeroth position since if were trying to find the 

> # eigth number, 8 % 8 = 0.

Reviewer: ```suggestion
# eighth number, 8 % 8 = 0.
```
